### PR TITLE
feat: added ability to define multiple email fields in alb and oauth2 stores

### PR DESCRIPTION
**Pull request by puehringer**
### Developer Checklist (Definition of Done)

**Issue**

- [x] All acceptance criteria from the issue are met
- [ ] Tested in latest Chrome/Firefox

**UI/UX/Vis**

- [ ] Requires UI/UX/Vis review
  - [ ] Reviewer(s) are notified (_tag assignees_)
  - [ ] Review has occurred (_link to notes_)
  - [ ] Feedback is included in this PR
  - [ ] Reviewer(s) approve of concept and design

**Code**

- [x] Branch is up-to-date with the branch to be merged with, i.e., develop
- [x] Code is cleaned up and formatted
- [x] Unit tests are written (frontend/backend if applicable)
- [ ] Integration tests are written (if applicable)

**PR**

- [x] Descriptive title for this pull request is provided (will be used for release notes later)
- [x] Reviewer and assignees are defined
- [x] Add type label (e.g., *bug*, *feature*) to this pull request
- [x] Add release label (e.g., `release: minor`) to this PR following [semver](https://semver.org/)
- [x] The PR is connected to the corresponding issue (via `Closes #...`)
- [x] [Summary of changes](#summary-of-changes) is written

### Summary of changes

- Sometimes, tokens look different across users, so defining an ordered array of fields to look in allows supporting multiple token layouts.
- Changed the way how env variables are loaded by default: now, all string values will be parsed as json (if possible). To avoid this, simply name the env variable (not the setting!) `..._as_string`. This will disable the json parsing.

### Additional notes for the reviewer(s)

-  
Thanks for creating this pull request 🤗

### CODE REVIEW

File: visyn_core/settings/model.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ class AlbSecurityStoreSettings(BaseModel):
     enable: bool = False
     cookie_name: str | None = None
     signout_url: str | None = None
-    email_token_field: str = "email"
+    email_token_field: str | list[str] = "email"
```
Comment (thinkh) [inline visyn_core/settings/model.py:47]: How do you configure a list via environment variables? Do you also need to parse it like JSON to dict?

File: visyn_core/settings/model.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ class OAuth2SecurityStoreSettings(BaseModel):
     cookie_name: str | None = None
     signout_url: str | None = None
     access_token_header_name: str = "X-Forwarded-Access-Token"
-    email_token_field: str = "email"
+    email_token_field: str | list[str] = "email"
```
Comment (thinkh) [inline visyn_core/settings/model.py:79]: Same here.

File: visyn_core/settings/model.py (hunk 1 above)
Comment (puehringer) [reply visyn_core/settings/model.py]: @thinkh Take a look at my latest [commit ](https://github.com/datavisyn/visyn_core/pull/81/commits/3989d2c23f6f6977fd52b13d825aa3b9e34635ae) where I add a `EnvSettingsSourceWithJSON` which does exactly that: convert every string into a dict if possible. If the env variable ends with `_as_string`, then no conversion happens (as sometimes we want `"2"` as actual string). 

Let me know what you think about this, the cool thing is it applies to all settings automatically, no need to add it anywhere else.

Comment (puehringer) [reply visyn_core/settings/model.py]: Another benefit: this exact way of parsing env variables exists in Pydantic 2, so if they have not fixed the nested env parsing (I have not checked with Pydantic 2), then this approach also works in v2: https://github.com/pydantic/pydantic-settings/blob/919a20b77527ecc1cd6eeb0a09ca22cc21486fb8/pydantic_settings/sources.py#L498-L542

**Review by thinkh — APPROVED** [review]
I've tested this with an additional list nested in the OE settings and configure it via environment variable and it worked in my tests. So I think we can merge this PR.

**Pull request merged**